### DISCUSSION TITLE
Why is the Modal component in ant design outside div tag which has id "root"?

### DISCUSSION THREAD

**Discussion by tungvu2210** [ant-design/ant-design#39796] [General]
I thought that every component in react app was mounted in the div tag has id "root". When my react app render, I found out that the entire app was mounted in the div tag has id "root" except the modal component in ant design. Can anyone explain to me why that is. Thanks for reading :smile:
![image](https://user-images.githubusercontent.com/66018884/209460370-6ca2e35f-ff9e-4c24-b3d2-12e9a09478f1.png)

**Comment by SignDawn** [comment]
父元素的 css 可能会影响子。
可参考 ReactDOM.createPortal

**Comment by SignDawn** [comment]
https://zh-hans.reactjs.org/docs/portals.html

**Comment by zombieJ** [comment]
Popup element append on the `document.body` by default to avoid cut off by `overflow: hidden` style. You can use ConfigProvider `getContainer` to modify default logic instead.

**Comment by phatify** [comment]
@zombieJ  same topic, but I have a question, what is the use of this TextArea?
![Screenshot 2022-12-29 225320](https://user-images.githubusercontent.com/70096611/209978116-7383cdf0-a7e1-4487-8ec8-ca67c27e3b0a.png)

**Reply by li2go** [reply]
cool!
